webapp/lib/model.g.dart

```diff
@@ -893,7 +899,9 @@ StreamSubscription<List<DocSnapshot>> listenForUpdates<T>(
       }
     }
     listener(added, modified, removed);
-  });
+  }, onError: (Object error) {
+    onErrorListener(error);
```

Comment:
if the `onErrorListener` is `null` (as it can be since it's optional in the `listenFor*` methods, then we should have a null check here:
```suggestion
    if (onErrorListener != null) onErrorListener(error);
```
(plus needs adding the stacktrace as per my other comment)

Reply:
Actually can you check if we can just pass the `onErrorListener` directly rather than wrapping it in the anonymous callback? something like `onError: onErrorListener`. And also what happens if it's `null` - I would hope that it just throws the error in the console as if there was no listener?

Reply:
Yes, we can be able to pass the `onErrorListener` directly. Nothing happens if it's `null`. I'm guessing it's because it only gets executed when there is an actual error.